contracts/account/extensions/ERC7579Modules/SocialRecoveryExecutor.sol

```diff
@@ -0,0 +1,462 @@
+        bytes32 digest = _getCancelRecoveryDigest(account, nonces(account));
+        _validateGuardianSignatures(account, guardianSignatures, digest);
+        _useNonce(account);
+
+        _cancelRecovery(account);
+    }
+
+    /// @notice Validates guardian signatures for a given digest
+    /// @dev Helper function for clients to validate signatures
+    /// @param account The ERC-7579 Account to validate signatures for
+    /// @param guardianSignatures Array of guardian signatures to validate
+    /// @param digest The digest to validate the signatures against
+    function validateGuardianSignatures(
+        address account,
+        GuardianSignature[] calldata guardianSignatures,
+        bytes32 digest
+    ) external view {
+        _validateGuardianSignatures(account, guardianSignatures, digest);
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////
+                                MODULE MANAGEMENT
+    //////////////////////////////////////////////////////////////////////////*/
+
+    /// @notice Adds a new guardian to the account's recovery configuration
+    /// @dev Only callable by the account itself
+    /// @param guardian Address of the new guardian
+    function addGuardian(address guardian) external {
+        _addGuardian(msg.sender, guardian);
+    }
+
+    /// @notice Removes a guardian from the account's recovery configuration
+    /// @dev Only callable by the account itself
+    /// @param guardian Address of the guardian to remove
+    function removeGuardian(address guardian) external {
+        _removeGuardian(msg.sender, guardian);
+    }
+
+    /// @notice Changes the number of required guardian signatures
+    /// @dev Only callable by the account itself
+    /// @param threshold New threshold value
+    function changeThreshold(uint256 threshold) external {
+        _changeThreshold(msg.sender, threshold);
+    }
+
+    /// @notice Changes the timelock duration for recovery
+    /// @dev Only callable by the account itself
+    /// @param timelock New timelock duration in seconds
+    function changeTimelock(uint256 timelock) external {
+        _changeTimelock(msg.sender, timelock);
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////
+                                VIEW FUNCTIONS
+    //////////////////////////////////////////////////////////////////////////*/
+
+    /// @notice Gets the current recovery status of an ERC-7579 Account
+    /// @param account The ERC-7579 Account to get the recovery status for
+    /// @return The current recovery status
+    function getRecoveryStatus(address account) public view virtual returns (RecoveryStatus) {
+        uint256 recoveryStart = _recoveryConfigs[account].recoveryStart;
+        if (recoveryStart == 0) {
+            return RecoveryStatus.NotStarted;
+        }
+        if (block.timestamp < recoveryStart + _recoveryConfigs[account].timelock) {
```

Comment:
What is the rational for storing the start, and doing that math when recovering the status versus storing the end (doing the add math when initiating a recovery) ?

I'm not sure if one is clearly better than the other, but we should documetn that choice with regard to the delay changing during the recovery.